### PR TITLE
ios: add remaining share types; process attachment in background on launch

### PR DESCRIPTION
**Pull request by levitatingpineapple**
## Share Extension PR1

- [x] Make comment background white/black
- [x] Revert translation to strings
- [x] Fix sending GIF (use `sendFile(...)` directly)
- [x] Send video message
- [x] Send link message
- [x] Send text message
- [x] Note, Text, Link should should not start event loop
- [x] Populate text field for editing

### CODE REVIEW

File: apps/ios/Shared/Views/Chat/ComposeMessage/ComposeLinkView.swift [hunk 1]
```diff
@@ -10,35 +10,6 @@ import SwiftUI
...
-                    }
-                }
-                cb(linkPreview)
-            }
-        } else {
-            cb(nil)
-        }
-    }
-}
-
```
Comment (levitatingpineapple) [inline apps/ios/Shared/Views/Chat/ComposeMessage/ComposeLinkView.swift:41]: Moved to `ImageUtils.swift`

File: apps/ios/Shared/Views/Helpers/ImagePicker.swift [hunk 2]
```diff
@@ -116,14 +115,12 @@ struct LibraryMediaListPicker: UIViewControllerRepresentable {
...
-                            let success = await makeVideoQualityLower(tempUrl, outputUrl: convertedVideoUrl)
-                            try? FileManager.default.removeItem(at: tempUrl)
-                            if success {
+                            let convertedVideoUrl = await transcodeVideo(from: tempUrl)
+                            if let convertedVideoUrl {
                                 _ = ChatModel.shared.filesToDelete.insert(convertedVideoUrl)
                                 let video = UploadContent.loadVideoFromURL(url: convertedVideoUrl)
                                 return cont.resume(returning: video)
                             }
-                            try? FileManager.default.removeItem(at: convertedVideoUrl)
```
Comment (levitatingpineapple) [inline apps/ios/Shared/Views/Helpers/ImagePicker.swift:126]: `transcodeVideo` will cleanup the url in case of failure

File: apps/ios/SimpleX SE/ShareView.swift [hunk 3]
```diff
@@ -73,13 +73,14 @@ struct ShareView: View {
...
                         }
+                        .disabled(model.isSendDisbled)
                     }
                 }
                 .frame(width: 28, height: 28)
                 .padding(6)
 
             }
-            .background(Color(.systemBackground.withAlphaComponent(0.5)))
+            .background(Color(.systemBackground))
```
Comment (levitatingpineapple) [inline apps/ios/SimpleX SE/ShareView.swift:83]: Make compose background solid white/black

File: apps/ios/SimpleXChat/ImageUtils.swift [hunk 4]
```diff
@@ -309,13 +310,47 @@ private func dropPrefix(_ s: String, _ prefix: String) -> String {
...
-        var actualTime = CMTimeMake(value: 0, timescale: 0)
-        if let image = try? generator.copyCGImage(at: CMTimeMakeWithSeconds(0.0, preferredTimescale: 1), actualTime: &actualTime) {
-            return (UIImage(cgImage: image), Int(duration.seconds))
+        do {
+            return (
+                UIImage(cgImage: try generator.copyCGImage(at: .zero, actualTime: nil)),
+                Int(duration.seconds)
+            )
+        } catch {
+            logger.error("Failed generate video preview \(error)")
```
Comment (levitatingpineapple) [inline apps/ios/SimpleXChat/ImageUtils.swift:353]: added logger error, to see if it fails

File: apps/ios/SimpleX SE/ShareViewController.swift [hunk 5]
```diff
@@ -26,14 +26,7 @@ class ShareViewController: UIHostingController<ShareView> {
...
         ShareModel.CompletionHandler.isEventLoopEnabled = false
-        if let extensionContext,
-           let item = extensionContext.inputItems.first as? NSExtensionItem {
-            model.completion = {
-                ShareModel.CompletionHandler.isEventLoopEnabled = false
-                extensionContext.completeRequest(returningItems: [item])
-            }
-            Task { await MainActor.run { model.item = item } }
-        }
+        model.setup(context: extensionContext!)
```
Comment (levitatingpineapple) [inline apps/ios/SimpleX SE/ShareViewController.swift:29]: Moved to shareModel. `extensionContext` is not @Sendable so the ref has to be passed before entering Task

File: apps/ios/SimpleXChat/ImageUtils.swift (hunk 4 above)
Comment (epoberezkin) [reply apps/ios/SimpleXChat/ImageUtils.swift:353]: please test in the app too

File: apps/ios/SimpleX SE/ShareModel.swift [hunk 6]
```diff
@@ -215,17 +208,18 @@ class ShareModel: ObservableObject {
...
+                if await !ch.isRunning { break }
             case let .chatItemStatusUpdated(_, ci):
                 guard isMessage(for: ci) else { continue }
                 if let (title, message) = ci.chatItem.meta.itemStatus.statusInfo {
+                    // `title` and `message` already localized and interpolated
                     return ErrorAlert(
-                        title: "\(title)",
-                        message: "\(message)"
+                        title: LocalizedStringKey(title),
+                        message: LocalizedStringKey(message)
```
Comment (epoberezkin) [inline apps/ios/SimpleX SE/ShareModel.swift:222]: why this change?

https://developer.apple.com/documentation/swiftui/localizedstringkey/init(_:)

This says that the parameter will be used as a localisation key, which is not what we want.
